app/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -124,6 +125,40 @@ static void kscan_direct_irq_callback_handler(const struct device *port, struct
 }
 #endif
 
+static gpio_flags_t kscan_gpio_get_flags(const struct gpio_dt_spec *gpio, bool active) {
+    if (((BIT(0) & gpio->dt_flags) == BIT(0))) { // Devicetree configured input ACTIVE_LOW
+        if (!active) {
+            return GPIO_ACTIVE_LOW | GPIO_PULL_UP;
+        }
+        return GPIO_ACTIVE_LOW;
+    } else { // Devicetree configured input ACTIVE_HIGH
+        if (!active) {
+            return GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN;
+        }
+        return GPIO_ACTIVE_HIGH;
+    }
+    LOG_ERR("Could not determine proper flags to set for pin %d", gpio->pin);
+    return 0;
+}
+
+static int kscan_inputs_set_flags(const struct kscan_gpio_list *inputs,
+                                  const struct gpio_dt_spec *active_gpio) {
+    gpio_flags_t updated_flags;
+    for (int i = 0; i < inputs->len; i++) {
+        updated_flags =
+            GPIO_INPUT | kscan_gpio_get_flags(&inputs->gpios[i], &inputs->gpios[i] == active_gpio);
+        LOG_DBG("Updated flags equal to: %d", updated_flags);
+
+        int err = gpio_pin_configure(inputs->gpios[i].port, inputs->gpios[i].pin, updated_flags);
```

Comment:
Do we need to resort to this instead of the "DT version" since that only takes *extra* flags, and will keep the pull set?

Reply:
That's actually why we need this one instead of the DT version. When we need to disable the pulls, the bitwise OR used in the DT version to apply the extra flags doesn't allow us to set the required bits to 0.

Reply:
I wonder, since the pull direction for a direct wire can be *directly* inferred from the active high/low setting, should we document/suggest that the pull direction not be set in the DT? That would seem to simplify the DT, and simplify this code quite a bit.

Thoughts?

Reply:
Definitely agree with you there. Just not sure where in the docs would be most appropriate to include this information. Perhaps we can update docs to include `toggle-mode` after #722 is merged?

Reply:
Actually, on second thought, if we leave the pins floating without a pull, I'm not sure if the toggle-switch will initialize with an active pin properly. I will test this on some hardware to verify if this is an error.

Reply:
I'm saying the init should calculate the appropriate pull as well and add it if not set via the extra_flags to the DT gpio call.

Reply:
Makes a lot more sense. *Currently* I haven't updated the init to do this yet. I'll start working on that part now.

Reply:
Can confirm things are working properly for both `GPIO_ACTIVE_LOW` and `GPIO_ACTIVE_HIGH`: proper pulls are calculated with those variables as the only flags in the DT.

Reply:
This can now be moved to `gpio_pin_configure_dt` it seems.